spec/rspec/core/option_parser_spec.rb

```diff
@@ -81,28 +83,101 @@ def generate_help_text
 
     %w[ -v --version ].each do |option|
       describe option do
-        it "prints the version and exits" do
+        it "prints the version and returns a zero exit code" do
           parser = Parser.new([option])
-          expect(parser).to receive(:exit)
 
           expect {
-            parser.parse
+            options = parser.parse
+            exit_code = options[:runner].call
+            expect(exit_code).to eq(0)
           }.to output("#{RSpec::Core::Version::STRING}\n").to_stdout
         end
       end
     end
 
+    %w[ -X --drb ].each do |option|
+      describe option do
+        let(:parser) { Parser.new([option]) }
+        let(:options) { double(:options) }
```

Comment:
It's not clear to me what this `let` is used for.  All the examples below I've looked at so far have a line like `options = ...` -- meaning they shadow this with a local variable and don't even use it.  Is this needed, or can it be moved into just the examples that need it?